src/modules/keepalive/keepalive_api.c

```diff
@@ -191,15 +196,14 @@ int ka_del_destination(str *uri, str *owner){
 		goto err;
 	}
 
+	lock_get(target->lock);
+	free_destination(target); // Lock is released and freed in this free function
 	if(!head){
 		LM_DBG("There isn't any head so maybe it is first \r\n");
 		ka_destinations_list->first = target->next;
-		free_destination(target);
-		ka_unlock_destination_list();
-		return 1;
+	} else {
+		head->next = target->next;
 	}
-	head->next = target->next;
-	free_destination(target);
 	ka_unlock_destination_list();
```

Comment:
You can't call free_destination(target) and after that use target

Reply:
Opps, thanks for the catch @linuxmaniac

Reply:
Fixed